components/Ui/Link/UiLink.vue

```diff
@@ -1,60 +1,83 @@
 <template>
-  <component
-    :is="isNuxtLink ? 'NuxtLink' : 'a'"
+  <NuxtLink
     v-track-click="segment"
-    :href="isAnchor && url"
-    :to="isNuxtLink && url"
-    :style="hasLink && 'cursor:pointer'"
-    :rel="isExternal && 'noopener'"
-    :target="isExternal ? '_blank' : undefined"
+    :external="computedExternal"
+    :target="computedTarget"
     :title="title"
+    :to="url"
     @click="emit('click')"
     @mouseenter="$emit('mouseenter')"
   >
     <slot />
-  </component>
+  </NuxtLink>
 </template>
 
 <script setup lang="ts">
 import { CtaClickedEventProp } from "~/types/segment";
 
+type NuxtLinkTarget =
+  | "_blank"
+  | "_parent"
+  | "_self"
+  | "_top"
+  | (string & {})
+  | null;
+
 interface Props {
+  external?: boolean;
   segment?: CtaClickedEventProp | undefined;
+  target?: NuxtLinkTarget;
   title?: string;
-  url?: string;
+  url: string;
 }
 
 const props = withDefaults(defineProps<Props>(), {
+  external: undefined,
   segment: undefined,
+  target: undefined,
   title: undefined,
-  url: "",
 });
 
 const emit = defineEmits(["click", "mouseenter"]);
 
-const isExternal = computed(() => {
-  return !!props.url && props.url.startsWith("http");
-});
+const config = useRuntimeConfig();
 
-const isInternal = computed(() => {
-  return !!props.url && props.url.startsWith("/");
+const isExternalFromSameDomain = computed<boolean>(() => {
+  return props.url.startsWith(config.public.siteUrl);
 });
 
-const isMail = computed(() => {
-  return !!props.url && props.url.startsWith("mailto");
+// Resolving link type (Based on https://github.com/nuxt/nuxt/blob/v3.6.3/packages/nuxt/src/app/components/nuxt-link.ts#L179)
+const isExternalFromDifferentDomain = computed<boolean>(() => {
+  if (isExternalFromSameDomain.value) {
+    return false;
+  }
+
+  // External prop is explicitly set
+  if (props.external) {
+    return true;
+  }
+
+  // When `target` prop is set, link is external
+  if (props.target && props.target !== "_self") {
+    return true;
+  }
+
+  return props.url === "" || urlHasProtocol(props.url);
 });
 
-const isIdAnchor = computed(() => {
-  return !!props.url && props.url.startsWith("#");
+const computedExternal = computed<boolean>(() => {
+  return isExternalFromSameDomain.value || isExternalFromDifferentDomain.value;
 });
 
-const hasLink = computed(() => !!props.url);
+const computedTarget = computed<NuxtLinkTarget | undefined>(() => {
+  if (typeof props.target !== "undefined") {
+    return props.target;
+  } else if (isExternalFromSameDomain.value) {
+    return "_self";
+  } else if (isExternalFromDifferentDomain.value) {
+    return "_blank";
+  }
 
-const isAnchor = computed(() => {
-  return (
-    isExternal.value || isMail.value || isIdAnchor.value || isInternal.value
-  );
+  return undefined;
```

Comment:
Not sure why `NuxtLinkTarget` has a `null` type but you return here an `undefined`

Reply:
The `target` prop in **NuxtLink** accepts `'_blank' | '_parent' | '_self' | '_top' | (string & {}) | null` and `undefined` since it's an optional Typescript property.

Here we're just trying to not interfere with their logic by passing the original value if one of our special cases doesn't need to rewrite it. The `computedTarget` function can pass both `null` and `undefined`.

Reply:
Understood , thank you!